sklearn/utils/estimator_checks.py

```diff
@@ -2120,6 +2123,118 @@ def check_classifiers_multilabel_representation_invariance(
     assert type(y_pred) == type(y_pred_list_of_lists)
 
 
+@ignore_warnings(category=FutureWarning)
+def check_classifiers_multilabel_format_output(name, classifier_orig):
+    """Check the output of the response methods for classifiers supporting
+    multilabel-indicator targets."""
+    classifier = clone(classifier_orig)
+    set_random_state(classifier)
+
+    n_samples, test_size, n_outputs = 100, 25, 5
+    X, y = make_multilabel_classification(n_samples=n_samples, n_features=2,
+                                          n_classes=n_outputs, n_labels=3,
+                                          length=50, allow_unlabeled=True,
+                                          random_state=0)
+    X = scale(X)
+
+    X_train, X_test = X[:-test_size], X[-test_size:]
+    y_train, y_test = y[:-test_size], y[-test_size:]
+    classifier.fit(X_train, y_train)
+
+    predict_method = getattr(classifier, "predict", None)
+    if predict_method is not None:
+        y_pred = predict_method(X_test)
+
+        # y_pred.shape -> y_test.shape with the same dtype
+        assert isinstance(y_pred, np.ndarray), (
+            f"{name}.predict is expected to output a NumPy array. Got "
+            f"{type(y_pred)} instead."
+        )
+        assert y_pred.shape == y_test.shape, (
+            f"{name}.predict output a NumPy array of shape {y_pred.shape} "
+            f"instead of {y_test.shape}."
+        )
+        assert y_pred.dtype == y_test.dtype, (
+            f"{name}.predict does not output the same dtype than the targets. "
+            f"Got {y_pred.dtype} instead of {y_test.dtype}."
+        )
+
+    predict_proba_method = getattr(classifier, "predict_proba", None)
+    if predict_proba_method is not None:
+        y_pred = predict_proba_method(X_test)
+
+        # y_pred.shape -> 2 possibilities:
+        # - list of length n_outputs of shape (n_samples, 2);
+        # - ndarray of shape (n_samples, n_outputs).
+        # dtype should be floating
+        if isinstance(y_pred, list):
+            assert len(y_pred) == n_outputs, (
+                f"When {name}.predict_proba returns a list, the list should "
+                f" be of length n_outputs and contain NumPy array. Got length "
+                f"of {len(y_pred)} instead of {n_outputs}."
```

Comment:
```suggestion
                f"When {name}.predict_proba returns a list, the list should "
                f" be of length n_outputs and contain NumPy arrays. Got length "
                f"of {len(y_pred)} instead of {n_outputs}."
```